src/Autofac.Extras.MvvmCross/AutofacMvxIocProvider.cs

```diff
@@ -146,23 +123,7 @@ public virtual void CallbackWhenRegistered<T>(Action action)
         /// </exception>
         public virtual void CallbackWhenRegistered(Type type, Action action)
         {
-            if (type == null)
-            {
-                throw new ArgumentNullException(nameof(type));
-            }
-
-            if (action == null)
-            {
-                throw new ArgumentNullException(nameof(action));
-            }
-
-            this.Container.ComponentRegistry.Registered += (sender, args) =>
-            {
-                if (args.ComponentRegistration.Services.OfType<TypedService>().Any(x => x.ServiceType == type))
-                {
-                    action();
-                }
-            };
+            this.provider.CallbackWhenRegistered(type, action);
```

Comment:
Can you confirm the argument checks are not required?

Reply:
Not possible unfortunately. We need a class that derives from `MvxSingleton<TInterface>`, which sets a static `Instance` property (with a private setter) and adds itself to a private static list of `Singletons`.  Furthermore, MvvmCross access `MvxSingleton<IMvxIoCProvider>` from many places.